### PR TITLE
refactor: remove getAsyncLogStream from SynchronousLogStream and rena…

### LINKED ISSUE #22832: Refactor distinction between LogStream and SyncLogStream
**Description**

With the unified flow control epic, the `LogStream` is now fully synchronous. This means the distinction between `SyncLogStream` (the test implementation of `LogStream`) and `LogStream` is not about async/sync, but rather that the test one has some extra helper functionality for writing tests.

The goal here is to refactor it so the distinction is clearer, and also remove any unnecessary fields/methods which may remain.

> [!Note]
> It's likely you can/should split up the resolution of this issue in multiple, bite-sized PRs.

### Acceptance criteria

- [ ] Remove reference to actor/scheduler in the `LogStream`
  - [ ] `LogStreamBuilder#withActorSchedulingService` method is removed
    - [ ] Calls to the method are removed
    - [ ] All implementation of this interface method are removed
  - [ ] Any dead code, fields, etc., due to the previous removal is also removed (e.g. `LogStreamBuilderImpl#actorSchedulingService` field, and its usage in the validate method)
- [ ] Remove `SyncLogStream#getAsyncLogStream` interface method
  - [ ] Remove implementations of this method
- [ ] Rename `SynchronousLogStreamWriter` to `BlockingLogStreamWriter`
  - [ ] Rename `SyncLogStream#newSyncLogStreamWriter` to `SyncLogStream#newBlockingLogStreamWriter`
- [ ] Rename `SyncLogStream` to `TestLogStream`
  - [ ] Rename related interfaces/classes appropriately (e.g. `SyncLogStreamBuilder` -> `TestLogStreamBuilder`)

### PR DESCRIPTION
**Pull request by entangled90**
## Description

refactor: remove getAsyncLogStream from SynchronousLogStream and rename SyncLogStream

Method getAsyncLogStream is removed from SynchronousLogStream.
Class SyncLogStream renamed to TestLogStream and delegate previously unsupported methods to field logStream.
Interface [SynchronousLogStream.java](https://github.com/camunda/camunda/pull/22948/files#diff-852ff4f97f701b4e29dd38d7735e022d1e92c46920e7b0162e879018468283fe) has been removed as it only had one implementation and only used in tests.

closes #22832

### CODE REVIEW

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/ProcessingExporterTransistor.java
Comment (npepinpe) [inline zeebe/engine/src/test/java/io/camunda/zeebe/engine/util/ProcessingExporterTransistor.java:48]: :wrench: Since we renamed the class too, might as well rename the field/variable.

```suggestion
          logStream.registerRecordAvailableListener(this::onNewEventCommitted);
```

File: zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/util/SynchronousLogStream.java
Comment (npepinpe) [inline zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/util/SynchronousLogStream.java:1]: :thought_balloon: I guess my issue was not clear enough :smile: 

:x: Please rename the interface as well. So I guess we go for `TestLogStream`, and `TestLogStreamImpl` for the implementation.

Comment (npepinpe) [reply zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/util/SynchronousLogStream.java:1]: To be honest, I would question why we need the interface in the first place for a test utility :thinking:

:question: Are there more implementations of `SynchronousLogStream`? Because if not, then I would propose maybe we just merge it with the implementation. I don't expect we would create more implementations of this interface honestly.

:x: And one last thing, could you also update the Javadocs? It still refers to `Synchronous methods` and the actor model.

**Review by npepinpe — APPROVED** [review]
:rocket: 

Please address all comments before merging, but unless you want to challenge them, I don't think we need another review pass here :+1:

File: zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/util/SynchronousLogStream.java
Comment (entangled90) [reply zeebe/logstreams/src/test/java/io/camunda/zeebe/logstreams/util/SynchronousLogStream.java:1]: Yes, it only has one implementation, therefore I merged it with the implementation.

**Pull request merged**
